realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java

```diff
@@ -592,6 +593,11 @@ public void run() {
         throw new RuntimeException("RealmQuery not supported yet.");
     }
 
+    @Override
+    public Completable from(Realm realm, Realm.Transaction transaction) {
+        return Completable.create(e -> realm.executeTransactionAsync(transaction, e::onComplete, e::onError));
```

Comment:
I'm a bit torn whether we should use `executeTransaction` or `executeTransactionAsync` here. If we use `executeTransaction()` we force people to manually use `subscribeOn`, but somehow that feels more idiomatic. The problem with using Realms internal async system is that it e.g. becomes impossible to swap schedulers when testing.

Did you have any reason for choosing this over the synchronous variant?

In any case, if we keep this, we need to update the doc to say it throws if called from a non-looper thread.

@nhachicha / @Zhuinden Do you have any thoughts?

Reply:
My thought is that this is a breaking change for anyone who implemented their custom `RxObservableFactory`.

But then again, it follows the https://github.com/realm/realm-java/issues/2126#issuecomment-196844283 proposition directly.

I think it should be possible to use `executeTransaction` vs `executeTransactionAsync` depending on whether the Realm is on a looper thread, which you can verify with `realm.isAutoRefresh()` or more-so the internal "capabilities" class should know whether this thread is able to use async transactions.

Reply:
If you use `executeTransaction ` then you need to make sure the Realm you're using will be on the same scheduler's Thread (`subscribeOn`). I think `executeTransactionAsync` is appropriate here